int64 B0_direction
int64 B0_speed
int64 B2_direction
int64 B2_speed
